File: php/src/Base/Commons/Dimensional.php
<?php

namespace Shapes\Base\Commons;

/**
 * # Summary of Measurable
 *
 * ``` php
 * <?php
 *
 * interface Measurable {
 *     public function getArea(): float;
 *     public function getPerimeter(): float;
 * }
 * ```
 */
interface Dimensional {
    /**
     * Print the area of the shape
     */
    public function area(): float;

    /**
     * Print the perimeter of the shape
     */
    public function perimeter(): float;
}

File: csharp/Src/Shapes/Base/Commons/Dimensional.cs
namespace Shapes.Base.Commons;

/// <summary>
/// Interface for objects that can be measured.
/// </summary>
public interface IDimensional {
    /// <summary>
    /// Calculates the area of the object.
    /// </summary>
    double Area();

    /// <summary>
    /// Calculates the perimeter of the object.
    /// </summary>
    double Perimeter();
}
